File: app/src/main/java/com/xmartlabs/template/BuildInfo.java
package com.xmartlabs.template;

import com.annimon.stream.Objects;

public class BuildInfo implements com.xmartlabs.base.core.model.BuildInfo {
  @Override
  public boolean isDebug() {
    return BuildConfig.DEBUG;
  }

  @Override
  public boolean isStaging() {
    return Objects.equals(BuildConfig.FLAVOR, BuildType.STAGING);
  }

  @Override
  public boolean isProduction() {
    return Objects.equals(BuildConfig.FLAVOR, BuildType.PRODUCTION);
  }
}

File: app/src/main/java/com/xmartlabs/template/model/BuildInfo.kt
package com.xmartlabs.template.model

import com.xmartlabs.template.BuildConfig
import com.xmartlabs.bigbang.core.model.BuildInfo as CoreBuildInfo

object BuildInfo : CoreBuildInfo {
  override val isDebug = BuildConfig.DEBUG
  override val isStaging = BuildConfig.FLAVOR == BuildType.STAGING.toString()
  override val isProduction = BuildConfig.FLAVOR == BuildType.PRODUCTION.toString()
}
